File: lab2/source/drop.cpp
#include "operations.h"
#include "factory.h"

class Drop : public Operations{
public:
    void Operation(std::stack<int> &stack, std::istream& fin) override {
        if(stack.empty()){
                throw std::runtime_error("stack is empty");
        }
        stack.pop();
    }
};

namespace {
    Operations* createDrop() {
        return new Drop;
    }
    bool x = OperationsFactory::getInstance()->RegisterOperation(std::string( "drop"), createDrop);
}

File: jaba/lab1/src/main/java/org/example/Drop.java
package org.example;

import java.io.OutputStream;
import java.util.Scanner;
import java.util.Stack;

public class Drop extends Operations{
    @Override
    public void Operation(Stack<Integer> stack, Scanner scanner, OutputStream s){
        emptyCheck(stack);
        stack.pop();
    }
}
